snap/validate.go

```diff
@@ -272,7 +272,7 @@ func validateSocketAddrPath(socket *SocketInfo, fieldName string, path string) e
 		return fmt.Errorf("invalid %q: %q should be written as %q", fieldName, path, clean)
 	}
 
-	if !(strings.HasPrefix(path, "$SNAP_DATA/") || strings.HasPrefix(path, "$SNAP_COMMON/")) {
+	if !(strings.HasPrefix(path, "$SNAP_DATA/") || strings.HasPrefix(path, "$SNAP_COMMON/") || strings.HasPrefix(path, "/run/user/0/$SNAP_NAME/")) {
```

Comment:
I think this should use `$SNAP_INSTANCE_NAME`

Reply:
I change it to XDG_RUNTIME_DIR, so there is no more need or SNAP_INSTANCE_NAME now